addons/mail/models/ir_attachment.py

```diff
@@ -38,15 +38,17 @@ def register_as_main_attachment(self, force=True):
             with contextlib.suppress(AccessError):
                 related_record.message_main_attachment_id = self
 
-    def _delete_and_notify(self):
+    def _delete_and_notify(self, message=None):
```

Comment:
That's a bit complicated code... calling "_message_update_content" on a model with a message calls "_delete_and_notify" on attachment to do model-based code (should not be here) and to write on the message (should probably be done in the main method) ... seems to me this code could be better cut.